include/okapi/api/chassis/controller/chassisController.hpp

```diff
@@ -92,213 +90,36 @@ class ChassisController : public ChassisModel {
    *
    * @param ishouldMirror whether turns should be mirrored
    */
-  virtual void setTurnsMirrored(bool ishouldMirror);
+  virtual void setTurnsMirrored(bool ishouldMirror) = 0;
 
   /**
    * Delays until the currently executing movement completes.
    */
   virtual void waitUntilSettled() = 0;
 
   /**
-   * Drive the robot forwards (using open-loop control). Uses velocity mode.
-   *
-   * @param ipower motor power
-   */
-  void forward(double ispeed) override;
-
-  /**
-   * Drive the robot in an arc (using open-loop control). Uses velocity mode.
-   * The algorithm is (approximately):
-   *   leftPower = forwardSpeed + yaw
-   *   rightPower = forwardSpeed - yaw
-   *
-   * @param iforwardSpeed speed in the forward direction
-   * @param iyaw speed around the vertical axis
+   * Interrupts the current movement to stop the robot.
    */
-  void driveVector(double iforwardSpeed, double iyaw) override;
+  virtual void stop() = 0;
 
   /**
-   * Drive the robot in an arc. Uses voltage mode.
-   * The algorithm is (approximately):
-   *   leftPower = forwardSpeed + yaw
-   *   rightPower = forwardSpeed - yaw
-   *
-   * @param iforwadSpeed speed in the forward direction
-   * @param iyaw speed around the vertical axis
-   */
-  void driveVectorVoltage(double iforwardSpeed, double iyaw) override;
-
-  /**
-   * Turn the robot clockwise (using open-loop control). Uses velocity mode.
-   *
-   * @param ipower motor power
-   */
-  void rotate(double ispeed) override;
-
-  /**
-   * Stop the robot (set all the motors to 0). Uses velocity mode.
-   */
-  void stop() override;
-
-  /**
-   * Drive the robot with a tank drive layout. Uses voltage mode.
-   *
-   * @param ileftSpeed left side speed
-   * @param irightSpeed right side speed
-   * @param ithreshold deadband on joystick values
-   */
-  void tank(double ileftSpeed, double irightSpeed, double ithreshold = 0) override;
```

Comment:
I'm assuming that the removal of these functions indicates that the user should call these from the internal ChassisModel, right? This change would break some of my code from last season but as long as it is still possible to have full control over the underlying ChassisModel then it would be find to work around this change.

Reply:
Users can call these from the internal ChassisModel, yes.